smfile_parser.py

```diff
@@ -48,56 +47,51 @@ def parse_sm(sm_file, new_file, output_dir):
     measure         = []
     measure_index   = 0
 
-    flag            = False
+    read_notes      = False
 
     with open(sm_file, encoding='ascii', errors='ignore') as f:
         for line in f:
-            if not flag:
+            line = line.rstrip('\n')
+            if not read_notes:
                 if line.startswith('#TITLE:'):
-                    step_dict['title']  = line.lstrip('#TITLE').lstrip(':').rstrip(';\n')
+                    step_dict['title']  = line.lstrip('#TITLE').lstrip(':').rstrip(';')
                 elif line.startswith('#BPMS:'):
                     if ',' in line:  # raises Exception if multiple BPMS detected
                         raise ValueError('Multiple BPMs detected')
-                    step_dict['BPM']    = float(line.lstrip('#BPMS:0.0').lstrip('0').lstrip('=').rstrip(';\n'))
+                    step_dict['bpm']    = float(split('=', line)[-1].rstrip(';'))
                 elif line.startswith('#OFFSET:'):
-                    step_dict['offset'] = float(line.lstrip('#OFFSET').lstrip(':').rstrip(';\n'))
-                elif line.startswith('#STOPS:') and line.rstrip('\n') != "#STOPS:;":
+                    step_dict['offset'] = float(split(':', line)[-1].rstrip(';'))
+                elif line.startswith('#STOPS:') and line != "#STOPS:;":
                     raise ValueError('Stop detected')
-
                 elif line.startswith('#NOTES:'):
-                    flag     = True
+                    read_notes     = True
```

Comment:
I realized now you can just split by the first `:` to get the type of metadata. Afterwards, you can parse the value based on what type of metadata it is.